web/web.go

```diff
@@ -416,23 +417,39 @@ func (h *Handler) Reload() <-chan chan error {
 func (h *Handler) Run(ctx context.Context) error {
 	level.Info(h.logger).Log("msg", "Start listening for connections", "address", h.options.ListenAddress)
 
-	listener, err := net.Listen("tcp", h.options.ListenAddress)
+	webListener, err := net.Listen("tcp", h.options.ListenAddress)
 	if err != nil {
 		return err
 	}
-	listener = netutil.LimitListener(listener, h.options.MaxConnections)
+	webListener = netutil.LimitListener(webListener, h.options.MaxConnections)
 
 	// Monitor incoming connections with conntrack.
-	listener = conntrack.NewListener(listener,
+	webListener = conntrack.NewListener(webListener,
 		conntrack.TrackWithName("http"),
 		conntrack.TrackWithTracing())
 
 	var (
-		m       = cmux.New(listener)
+		m       = cmux.New(webListener)
 		grpcl   = m.Match(cmux.HTTP2HeaderField("content-type", "application/grpc"))
```

Comment:
should we not keep  this one  if they are different?

Reply:
it's overridden at line 450 by the new one based on the new grpcListener